File: main/src/main/java/com/sedmelluq/lavaplayer/core/http/HttpInterfaceManager.java
package com.sedmelluq.lavaplayer.core.http;

import java.io.Closeable;

/**
 * A thread-safe manager for HTTP interfaces.
 */
public interface HttpInterfaceManager extends ExtendedHttpConfigurable, Closeable {
  /**
   * @return An HTTP interface for use by the current thread.
   */
  HttpInterface getInterface();
}

File: main/src/main/java/lavaplayer/tools/io/HttpInterfaceManager.kt
package lavaplayer.tools.io;

import lavaplayer.tools.http.ExtendedHttpConfigurable;

import java.io.Closeable;

/**
 * A thread-safe manager for HTTP interfaces.
 */
interface HttpInterfaceManager : ExtendedHttpConfigurable, Closeable {
    /**
     * @return An HTTP interface for use by the current thread.
     */
    fun get(): HttpInterface;
}
